p32/src/lib.rs

```diff
@@ -0,0 +1,359 @@
+use crate::TransferFundsError::{
+    ReceiverNotExistsError, SenderNotEnoughBalance, SenderNotExistsError,
+};
+
+pub struct User {
+    name: String,
+    credit_line: u64,
+    balance: i64,
+}
+
+impl User {
+    pub(crate) fn new(name: String, credit_line: u64, balance: i64) -> Self {
+        User {
+            name,
+            credit_line,
+            balance,
+        }
+    }
+}
+
+pub struct Bank {
+    users: Vec<User>,
+    name: String,
+    credit_interest: u64,
+    debit_interest: u64,
+}
+
+impl Bank {
+    pub fn merge_bank(&mut self, mut other: Bank) -> Bank {
+        let mut merged_users: Vec<User> = vec![];
+        // TODO: is there a function call chain to zip by a given property?
+        // Instead of:
+        // self.users.iter().zip(merged_users.iter_mut()).for_each(|(user, merged_user)| {})
+        // Try:
+        // https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.chunk_by
+        //
+        // self.users.chunk_by(|user| user.name);
+        // Not working because the lambda needs to be a predicate
+        //
+        // Try
+        // v.into_iter().map(|n| (n, f(n))).collect();, from https://www.reddit.com/r/rust/comments/18m24wb/how_to_convert_vec_to_hashmap/
+        // let users_by_name = self.users.into_iter().map(|n| (n.name, n)).collect();
+        // Compilation error = Cannot move
+
+        for user in &mut self.users {
+            let maybe_overlapping_user = other.users.iter_mut().find(|x| x.name == user.name);
+            let mut balance = user.balance;
+            if let Some(overlapping_user) = maybe_overlapping_user {
+                balance += overlapping_user.balance;
+                overlapping_user.balance = 0;
+            }
+            other.users.retain(|x| x.name != user.name);
+            merged_users.push(User::new(user.name.clone(), user.credit_line, balance));
+        }
+
+        for non_overlapping_user in &other.users {
+            merged_users.push(User::new(
+                non_overlapping_user.name.clone(),
+                non_overlapping_user.credit_line,
+                non_overlapping_user.balance,
+            ));
+        }
+
+        drop(other);
+
+        Bank {
```

Comment:
You don't need to return a merged bank. You should modify in place the bank into which we merge the second bank (i.e. via `&mut self`).